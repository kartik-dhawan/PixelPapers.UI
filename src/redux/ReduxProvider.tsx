"use client"

/**
 *
 * This component acts as a parent for client side components that require redux store
 * Wrap the component with this 'Redux Provider' to use state management on client side
 *
 * the updates made on the client side in the store will not be reflected in the server side components
 *
 * SERVER SIDE:
 * - Dispatching: store.dispatch(functionName(payload))
 * - Getting data from store: store.getState().sliaceName.stateName
 *
 * CLIENT SIDE:
 * - Dispatching: dispatch(functionName(payload)) [useDispatch()]
 * - Getting data from store: useSelector((state)=>state.slice.statename)
 *
 * Ref: https://www.youtube.com/watch?v=dRLjoT4r-jc
 */

import { Provider } from "react-redux"
import { store } from "./store"
import { ChildrenProps } from "@/utils/interfaces"

const ReduxProvider = ({ children }: ChildrenProps) => {
  return <Provider store={store}>{children}</Provider>
}

export default ReduxProvider
