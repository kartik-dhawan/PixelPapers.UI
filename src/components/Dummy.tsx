"use client"

import { RootType } from "@/redux/store"
import { useSelector } from "react-redux"

const Dummy = () => {
  const { count } = useSelector((state: RootType) => state.countSlice)

  console.log(count)
  return <div>Hello world</div>
}

export default Dummy
