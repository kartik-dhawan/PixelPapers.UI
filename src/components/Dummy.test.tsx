import "@testing-library/jest-dom"
import renderer from "react-test-renderer"
import Dummy from "./Dummy"
import ReduxProvider from "@/redux/ReduxProvider"

describe("Experience section", () => {
  it("-- should match the snapshot", () => {
    const tree = renderer.create(
      <ReduxProvider>
        <Dummy />
      </ReduxProvider>,
    )
    expect(tree).toMatchSnapshot()
  })
})
