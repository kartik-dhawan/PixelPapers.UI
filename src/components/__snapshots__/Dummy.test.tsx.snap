// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Experience section -- should match the snapshot 1`] = `
<div>
  Hello world
</div>
`;
